src/resolvers/Mutation/updateAdvertisement.ts

```diff
@@ -0,0 +1,83 @@
+import type { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { Advertisement, User } from "../../models";
+import { errors, requestContext } from "../../libraries";
+import {
+  ADVERTISEMENT_NOT_FOUND_ERROR,
+  USER_NOT_FOUND_ERROR,
+  INPUT_NOT_FOUND_ERROR,
+  END_DATE_VALIDATION_ERROR,
+  START_DATE_VALIDATION_ERROR,
+} from "../../constants";
+
+export const updateAdvertisement: MutationResolvers["updateAdvertisement"] =
+  async (_parent, args, _context) => {
+    const { _id, ...otherFields } = args.input;
+
+    //If there is no input
+    if (Object.keys(otherFields).length === 0) {
```

Comment:
In graphql you can provide `null` for the fields that are nullable in schema, this check wouldn't check against that.

Also, what fields should the user be allowed to update? What should be done if the user tries to update a field to a value that is not intended? One example would be making the `organizationId` `null`. This is unintended.

Think of all the possible scenarios possible in a real world workflow, and then express logic in the resolver accordingly.